src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts

```diff
@@ -0,0 +1,236 @@
+      });
+
+      it('should return null if all questions have been answered', () => {
+        spyOn(component, 'everyQuestionHasOutcome').and.returnValue(true);
+        spyOn(component, 'incompleteVehicleChecks');
+        const result = component.validateVehicleChecks(null);
+        expect(result).toEqual(null);
+      });
+    });
+
+    describe('invalid', () => {
+
+      beforeEach(() => {
+        const formBuilder: FormBuilder = new FormBuilder();
+        component.formGroup = formBuilder.group({
+          vehicleChecksSelectQuestions: null,
+        });
+        component.formControl = formBuilder.control({});
+      });
+
+      describe('when form is dirty', () => {
+        it('should return TRUE if the form control is invalid', () => {
+          component.formControl.markAsDirty();
+          component.formControl.setErrors({ vehicleChecks: false });
+          const result = component.invalid;
+          expect(result).toEqual(true);
+        });
+
+        it('should return FALSE if the form control is valid', () => {
+          component.formControl.markAsDirty();
+          const result = component.invalid;
+          expect(result).toEqual(false);
+        });
+      });
+
+      describe('when form is NOT dirty', () => {
+        it('should return FALSE if the form control is invalid', () => {
+          component.formControl.markAsPristine();
+          const result = component.invalid;
+          expect(result).toEqual(false);
+        });
+      });
+    });
+
+    describe('ngOnChanges', () => {
+      it('should add the form control', () => {
+        const formBuilder: FormBuilder = new FormBuilder();
+        component.formGroup = formBuilder.group({
+          vehicleChecksSelectQuestions: null,
+        });
+        spyOn(component, 'everyQuestionHasOutcome').and.returnValue(true);
+        component.ngOnChanges();
+        const result = component.formGroup.contains('vehicleChecksSelectQuestions');
+        expect(result).toEqual(true);
+      });
+
+      it('should validate the vehicle checks', () => {
+        const formBuilder: FormBuilder = new FormBuilder();
+        component.formGroup = formBuilder.group({
+          vehicleChecksSelectQuestions: null,
+        });
+        spyOn(component, 'everyQuestionHasOutcome').and.returnValue(true);
+        spyOn(component, 'validateVehicleChecks');
+        component.ngOnChanges();
+        expect(component.validateVehicleChecks).toHaveBeenCalled();
+      });
+
+      it('should patch the form control value', () => {
+        const formBuilder: FormBuilder = new FormBuilder();
+        component.formGroup = formBuilder.group({
+          vehicleChecksSelectQuestions: null,
+        });
+        component.formControl = formBuilder.control({});
+        component.ngOnChanges();
+        expect(component.formControl.value).toEqual('Select questions');
+      });
+    });
+
+    describe('DOM', () => {
```

Comment:
I know you didn't add it, it's elsewhere in the code. But can we get rid of these?

Reply:
You want me to delete the `DOM` tests?

Reply:
deleted